### PR TITLE
Vmo 2623 tidy new code to remove ts errors

### PR DESCRIPTION
**Pull request by safydy-r**
[Remaining // ts-ignore in `src/` folder]
- in TreeBuilderToolbar.vue & flow.ts: related with IBlock `platform_metadata`, I think it's ok for now

For dev info:
I found a solution to use mixins with vue-class based components:
- it's important to turn mixins files into `.ts` first (I don't know why js files won't work)
- then follow this pattern https://class-component.vuejs.org/guide/extend-and-mixins.html

### CODE REVIEW

File: src/components/interaction-designer/resource-editors/ResourceEditor.vue
Comment (safydy-r) [inline src/components/interaction-designer/resource-editors/ResourceEditor.vue:117]: turned to vue-class based

File: src/components/interaction-designer/resource-editors/ResourceEditor.vue
Comment (safydy-r) [inline src/components/interaction-designer/resource-editors/ResourceEditor.vue:130]: Note this guys, you may need it for future usage ;)

File: src/components/interaction-designer/toolbar/TreeBuilderToolbar.vue
Comment (safydy-r) [inline src/components/interaction-designer/toolbar/TreeBuilderToolbar.vue:184]: turned into vue-class based

File: src/store/flow/block-types/MobilePrimitives_SelectOneResponseBlockStore.ts
Comment (safydy-r) [inline src/store/flow/block-types/MobilePrimitives_SelectOneResponseBlockStore.ts:54]: As we don't have access to rootState  from Mutation & this.state.flow throws ts exception, I turned this into Actions

File: src/components/interaction-designer/block-types/ConsoleIO_PrintBlock.vue
Comment (jcbashdown) [inline src/components/interaction-designer/block-types/ConsoleIO_PrintBlock.vue:59]: Nice. I almost did this in my branch but decided not to mess with `@/lib/filters/lang`. Good to finally get it done. I notice a few places we will use the old `mixins: [lang]`. Should we update those?

File: src/store/flow/block.ts
Comment (jcbashdown) [inline src/store/flow/block.ts:12]: Noticed you also do this in 2296. I suppose this will shake out in the conflict resolution...

File: src/store/flow/flow.ts
Comment (jcbashdown) [inline src/store/flow/flow.ts:82]: How come this is no longer necessary?

File: tsconfig.json
Comment (jcbashdown) [inline tsconfig.json:14]: What effect does this have exactly?

**Review by jcbashdown — CHANGES_REQUESTED** [review]
Looks good but I have some concerns about merge order and I've made a couple of comments about small possible issues/changes. Let me know what you think.

File: src/components/interaction-designer/block-types/ConsoleIO_PrintBlock.vue
Comment (safydy) [reply src/components/interaction-designer/block-types/ConsoleIO_PrintBlock.vue:59]: Changing the old approach `mixins: [lang]` requires we turn the component into full vue class, which could be time consuming and is not very important for now. I'd propose we create a ticket for Beta version to turn all component into class based.

File: src/store/flow/block.ts
Comment (safydy) [reply src/store/flow/block.ts:12]: Thanks for noticing. I think it will be a conflict. Anyway, any change in this line would create a conflict, so I will leave this here.

File: src/store/flow/flow.ts
Comment (safydy) [reply src/store/flow/flow.ts:82]: Um, IFlowState doesn't have `builder` prop, therefore this is always a TS error. So I moved this out of this mutation https://github.com/FLOIP/flow-builder/pull/55/files#diff-3db5c56652da6d4f7b33c23e432a38b9b73d0b30d47df648929798e4c82d498bR334.

But when thinking longer, I found a solution:
- instead of having a mutation, which doesn't give us access to `commit` or `dispatch`
- I changed it into action. Then I changed this line into `commit('builder/activateBlock', { blockId: null }, { root: true })`

Thanks for noticing

File: tsconfig.json
Comment (safydy) [reply tsconfig.json:14]: Well, this is more like a proposal than a must have change.

here is the definition from the official doc https://www.typescriptlang.org/tsconfig#strictPropertyInitialization
> TypeScript will raise an error when a class property was declared but not set in the constructor.

Why do we need it?
Our current code has TS error like this
```
ERROR in /home/safydy/Documents/flow-builder/src/components/interaction-designer/resource-editors/ResourceEditor.vue(131,29):
131:29 Property 'block' has no initializer and is not definitely assigned in the constructor.
    129 |   })
    130 | export class ResourceEditor extends mixins(FlowUploader, Permissions, Routes, Lang) {
  > 131 |   @Prop({ required: true }) block: IBlock
           |                                            ^
    132 | 
    133 |   @Prop({ required: true }) flow: IFlow
    134 | 
ERROR in /home/safydy/Documents/flow-builder/src/components/interaction-designer/resource-editors/ResourceEditor.vue(133,29):
133:29 Property 'flow' has no initializer and is not definitely assigned in the constructor.
    131 |   @Prop({ required: true }) block: IBlock
    132 | 
  > 133 |   @Prop({ required: true }) flow: IFlow
           |                                            ^
    134 | 
    135 |   @Prop() resource: IResourceDefinition
```

The solution is to provide initial value `@Prop({ required: true, default: {} }) block: IBlock = {} as IBlock`, but it's too repetitive to my POV. 
As I never had an issue without having initialized a prop in the past, I thought we could ignore this kind of error from the tsconfig. What do you think?

File: src/store/flow/flow.ts
Comment (safydy) [reply src/store/flow/flow.ts:82]: put it back https://github.com/FLOIP/flow-builder/pull/55/files#diff-eea673c657b7edcdbf00ab176fcc8ec0a84d21e8a8dfa9ff8a014c7510a144a0R134

File: src/components/interaction-designer/block-types/ConsoleIO_PrintBlock.vue
Comment (jcbashdown) [reply src/components/interaction-designer/block-types/ConsoleIO_PrintBlock.vue:59]: Sounds good. Let's leave it

File: src/store/flow/flow.ts
Comment (jcbashdown) [reply src/store/flow/flow.ts:82]: Okay, cool. I think that should work

File: tsconfig.json
Comment (jcbashdown) [reply tsconfig.json:14]: How about this instead? 

https://github.com/kaorun343/vue-property-decorator/issues/81#issuecomment-365941684

The issue with this approach is that I think it is hiding that resource and label are not required/don't have a default. We should make sure we set these in all cases.

Comment (jcbashdown) [reply tsconfig.json:14]: BTW. I suggest we don't prioritise this ticket for the next sprint while we still have functional changes to do. We won't be able to merge until flow lifecycle is merged anyway.

Comment (safydy) [reply tsconfig.json]: @jcbashdown , thanks for the suggestion. Let's adopt the `definite assignment assertion` then. I have remove the tsconfig `"strictPropertyInitialization": false`.
About priority, yes we should prioritize flow lifecycle as discussed, but upcoming functionalities (eg: Validation API tickets: vmo-3710, vmo-3704, etc) were branched out of this to avoid headache on conflict resolution. Our plan is to merge this asap once flow lifecycle is merged.

**Review by jcbashdown — APPROVED** [review]
Looks good! Hoping to merge lifecycle this morning and unblock finishing this

File: src/store/flow/block-types/Core_RunFlowBlockStore.ts
Comment (bzabos) [inline src/store/flow/block-types/Core_RunFlowBlockStore.ts:17]: Interesting! How come this syntax worked and the rocket (`=>`) shorthand wouldn't? 🙀

**Review by bzabos — APPROVED** [review]
Wow, thanks! Looks good to me.

File: src/store/flow/block-types/Core_RunFlowBlockStore.ts
Comment (safydy-r) [reply src/store/flow/block-types/Core_RunFlowBlockStore.ts:17]: No idea, sorry :D

**Pull request merged**
